osu.Game/Screens/Play/SongProgressGraph.cs

```diff
@@ -6,43 +6,207 @@
 using System.Diagnostics;
 using osu.Game.Rulesets.Objects.Types;
 using osu.Game.Rulesets.Objects;
+using osu.Framework.Timing;
+using osu.Game.Beatmaps.Timing;
 
 namespace osu.Game.Screens.Play
 {
     public class SongProgressGraph : SquareGraph
     {
+        private List<double> strains = new List<double>();
+
+        public List<double> Strains
+        {
+            get { return strains; }
+            set
+            {
+                for(int x = 0; x < value.Count; x++)
+                {
+                    if (x == 0)
+                    {
+                        strains.Add(value[x]);
+                        strains.Add(value[x]);
+                        Values.Add(value[x]);
+                        Values.Add(value[x]);
+                    }
+                    else
+                    {
+                        strains.Add(value[x]);
+                        Values.Add(value[x]);
+                    }
+                }
+            }
+        }
+
         private IEnumerable<HitObject> objects;
 
         public IEnumerable<HitObject> Objects
         {
+            get { return objects; }
             set
             {
                 objects = value;
 
-                const int granularity = 200;
-                Values = new int[granularity];
-
                 if (!objects.Any())
-                    return;
+                {
+                    for (int x = 0; x < Strains.Count; x++)
+                    {
+                        Strains[x] = 0;
+                    }
+                }
+                else
+                {
+                    if (Strains.Count==0)
+                    {
+                        Strains.Add(1);
+                        Update();
+                        return;
+                    }
+                }
 
-                var firstHit = objects.First().StartTime;
-                var lastHit = objects.Max(o => (o as IHasEndTime)?.EndTime ?? o.StartTime);
+                var values = new List<int>();
 
-                if (lastHit == 0)
-                    lastHit = objects.Last().StartTime;
+                for (int x = 0; x < strains.Count; x++)
+                {
+                    values.Add(0);
+                }
 
-                var interval = (lastHit - firstHit + 1) / granularity;
+                var startOfLists = (objects.First().StartTime - objects.First().StartTime % (strainStep * audioClock.Rate))/(strainStep * audioClock.Rate);
+                if (strainStep==1)
```

Comment:
Add spaces between operator and operands, apply for the following code, too many occurrences